lib/Docker/Registry/Azure.pm

```diff
@@ -0,0 +1,31 @@
+package Docker::Registry::Azure;
+  use Moose;
+  extends 'Docker::Registry::V2';
+
+  has '+url' => (lazy => 1, default => sub {
```

Comment:
You could use `DockerRegistryURI` for this, see Auth::Gitlab with on how to do this.